### PR TITLE
Add sim_plots testing with Dash testing (Dash duo)

### PR DESCRIPTION
**Pull request by calina-c**
Work on #1209.

### What was done:
- add dependencies for dash duo end-to-end-testing, removed an unused dependency from the codebase
- integration testing for sim_plots dashboard:
     - test all tabs show up
     - test clicking on tabs display each figure/figure group as needed
     - [ ] PENDING: test clicking on bars in var_imps chart toggles contour/line plot -> see below for details
     - improve coverage for various modules -> see below for details

### Varimps issue
I have to work with test parameters to ensure we get a contour plot and varimps can have both variables displayed. However, there is also an UX issue. Due to resizing and adding tabs, the tooltip of the varimps bar covers the bar itself and prevents clicking. We need to fix that first and I'll talk to @KatunaNorbert about it + create a follow-up issue. In the meantime, I propose we merge this branch and continue integration tests for other dashboards.

### Coverage changes
`dash_plots/callbacks.py`: 31% -> 75% (probably full coverage after fixing varimps and adding test for click changes)
`dash_plots/view_elements`: 97% -> 100%
`sim_dash.py`: 80% -> 75% (removed some rows and left as a simple wrapper)
`sim_engine.py`: 99%->98%
`sim_plotter.py`: 36% -> 80% (probably full after fixing varimps and adding test for click changes)
`sim_state.py`: 96% remained the same

### CODE REVIEW

**Comment by calina-c** [comment]
Screenshot 2024-06-18 at 09 41 23

UX issue that prevents me from clicking the bar is attached.

**Comment by calina-c** [comment]
Update ref. the var_imps issue:
- the visual click issue is minor and we'll see what we can do, but that will surely be a separate task
- I don't think the effort into adding and testing the varimps bar click is warranted, at least not right now. It's very difficult due to the following factors:
    1. there is no way to manipulate the test data in sim_engine.py to balance variable importances so that multiple bars appear. And if just a single bar appears, all bars are considered "selected" so there is no way to distinguish through the "selected vars" if the click was done
    2. no way to detect the redraw of the line/contour plot itself. We could try some "fireworks" like giving it a unique id to fix this, but still, see (3)
    3. It is incredibly difficult to perform the click. Plotly figures are svgs, so we can not detect individual elements with selectors. We could try offsets of x/y from the corner of the figure but since the selenium backend uses who-knows-what resolution it will take a very long time to figure out and it's possible it will behave badly on other setups.
    
TL;DR: I think we can close the sim_plots part of this task. @trentmc, what do you think?

**Comment by trentmc** [comment]
> TL;DR: I think we can close the sim_plots part of this task. @trentmc, what do you think?

OK.

**Review by trentmc — APPROVED** [review]
